runtime/src/main/java/org/corfudb/protocols/logprotocol/StreamedLogData.java

```diff
@@ -1,48 +1,142 @@
 package org.corfudb.protocols.logprotocol;
 
 import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
 import lombok.Getter;
 import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.ICorfuPayload;
 import org.corfudb.protocols.wireprotocol.ILogData;
 import org.corfudb.runtime.CorfuRuntime;
 
 import javax.annotation.Nonnull;
-import java.util.EnumMap;
-import java.util.Map;
-import java.util.UUID;
-import java.util.function.BiConsumer;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Temporary class for transition to log data.
  *
  * Created by mwei on 4/6/17.
  */
-public class StreamedLogData implements ILogData {
+public class StreamedLogData implements ILogData, ICorfuPayload<StreamedLogData>{
 
+    /** A map of stream data. */
     @Getter
     public final @Nonnull
-    Map<UUID, StreamData> entryMap;
+    Map<UUID, StreamData> streamDataMap;
 
-    public StreamedLogData(Map<UUID, StreamData> entryMap) {
-        this.entryMap = entryMap;
+    /** The metadata for this log entry. */
+    @Getter
+    final EnumMap<LogUnitMetadataType, Object> metadataMap;
+
+    /** The serialized form, cached during writing only
+     * and auto-released by the handle.
+     */
+    protected ByteBuf serializedForm;
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<UUID> getStreams() {
+        return streamDataMap.keySet();
     }
 
-    public void getSerializedForm(BiConsumer<Object, ByteBuf> serializer, ByteBuf b) {
-        // go through each stream data and serialize it.
+    /** {@inheritDoc} */
+    @Override
+    public boolean containsStream(UUID stream) {
+        return streamDataMap.containsKey(stream);
     }
 
+    /** {@inheritDoc} */
     @Override
-    public Object getPayload(CorfuRuntime t) {
-        return null;
+    public Map<UUID, Long> getBackpointerMap() {
```

Comment:
I can't imagine why we would need this method in the future; but perhaps we need it in the interim?